src/redux/selectors/appMessagesSelectors.js

```diff
@@ -0,0 +1,73 @@
+import { createSelector } from 'reselect';
+import { rhsmApiTypes } from '../../types';
+/**
+ * Selector cache.
+ *
+ * @private
+ * @type {{data: {object}}}
+ */
+const selectorCache = { data: {} };
+
+/**
+ * Return a combined state, props object.
+ *
+ * @private
+ * @param {object} state
+ * @param {object} props
+ * @returns {object}
+ */
+const statePropsFilter = (state, props = {}) => ({
+  reportCapacity: state.graph?.reportCapacity?.[props.productId],
+  viewId: props.viewId,
+  productId: props.productId
+});
+
+/**
+ * Create selector, transform combined state, props into a consumable object.
+ *
+ * @type {{appMessages: {cloudigradeMismatch: boolean}}}
+ */
+const selector = createSelector([statePropsFilter], data => {
+  const { viewId = null, productId = null, reportCapacity = {} } = data || {};
+  const appMessages = {
+    cloudigradeMismatch: false
+  };
+
+  const cache = (viewId && productId && selectorCache.data[`${viewId}_${productId}`]) || undefined;
```

Comment:
note: the cache is specific toward viewId and productId making the banner message shut down between architectures. if we want to say the banner is activated across a products architectures we should remove `...${productId}` from the cache key/string